UI/src/elements/lsmb-country.js

```diff
@@ -0,0 +1,72 @@
+/** @format */
+
+import { LsmbBaseInput } from "@/elements/lsmb-base-input";
+import { useCountriesStore } from "@/store/countries";
+
+const Select = require("dijit/form/Select");
+const registry = require("dijit/registry");
+
+export class LsmbCountry extends LsmbBaseInput {
+    store = null;
+
+    widgetWrapper = null;
+
+    _boolAttrs() {
+        return ["default_blank"];
+    }
+
+    _widgetRoot() {
+        if (this.widgetWrapper) {
+            return this.widgetWrapper;
+        }
+        this.widgetWrapper = document.createElement("span");
+        this.appendChild(this.widgetWrapper);
+
+        return this.widgetWrapper;
+    }
+
+    async connectedCallback() {
+        let props = this._collectProps();
+
+        if (!this.store) {
+            /* Instantiate *and* initialize store before assigning to
+             * the object property to avoid a race condition.
+             */
+            const store = useCountriesStore();
+            await store.initialize();
+            this.store = store;
+        }
+
+        const options = this.store.items.map((element) => {
+            return { label: element.name, value: element.code };
```

Comment:
you may be able to skip this "renaming" operation by setting the `idProperty` and `labelAttr`  to `code` and `name` respectively. Then you can return the elements as-is.

Reply:
Unfortunately, as far as I can see, the dojo Select widget doesn't support `idProperty` and `labelAttr` attributes - I've tested it and looked at the docs here: https://dojotoolkit.org/reference-guide/1.10/dijit/form/Select.html

Am I missing something?

Reply:
I'm using https://dojotoolkit.org/api/ ; but, no, I don't think you missed something. I genuinely thought that the `idProperty` existed. (A major shortcoming it doesn't, really, but that's beyond the scope of this PR.)